PWGHF/HFC/TableProducer/correlatorHfeHadrons.cxx

```diff
@@ -135,7 +135,7 @@ struct HfCorrelatorHfeHadrons {
       registry.fill(HIST("hptElectron"), ptElectron);
       for (const auto& hTrack : tracks) {
 
-        if (hTrack.globalIndex() == eTrack.globalIndex())
+        if (hTrack.globalIndex() == eTrack.gTrackId())
```

Comment:
It is not clear why globalIndex() is used for hadron track and gTrackId() is used for electron track. This can be error prone. If there is an issue with globalIndex function in the electron table, I would suggest to investigate that and fix it.

Reply:
Dear deepa, For hadron, information we are fetching directly from the track table.  First index of the track table is the global index or track id.
Whereas , in  case of electrons, we have saved  separately in a  table. The first index of the electron table is no more the global index or you can say not the track id.
So I separately saved track id of  electron in the electron table which is the original global index from the track table. So global index of hadrons and  track id of electrons are same.
since , we have saved electron separately  in a table , first index of the table  is basically local electron index , no more global index.
Yes I agree both  functions( gTrackId() , globalIndex()) are different, however the information coming from the function are same .
Maximum what i can think in line 138 in correlatorHfeHadrons.cxx task : 
hadronId = hTrack.globalIndex();
electronId = eTrack.gTrackId();
   if (hadronId  == electronId)
          Continue;
I hope now things are clear to you. If you have any suggestions please let me know.

Reply:
ok. I am ok if you proceed with your proposal. I dont have any alternative suggestion. Maybe @vkucera has some suggestion. Else I will approve the PR

Reply:
@rashigupt I understand your point but @deepathoms also as a point. The track ID that you store in the electron table is still an index of the track table and therefore it looks natural to store it in an index column. So I don't understand the change of the column definition in your PR.

Reply:
Also `GTrackId` is not really a self-explanatory name.

Reply:
When I used eTrack.globalIndex() in the correlation table, it returned the electron index instead of the track ID.

Reply:
Which is what `globalIndex()` is supposed to do. You have to get the index from the column where you saved it.

Reply:
Thanks to Deepa and Vit, I used the correct electron ID (trackID) without making changes to the column definition in the table.h file.